### PR TITLE
WIP: Bump to Kubernetes v1.6.1

### PR DESCRIPTION
**Pull request by mumoshu**
As etcd3 support is already introduced via #417, it was (ideally) a matter of running `bump-version` and then run E2E tests against a newly created kube-aws cluster with k8s 1.6.1.

- [x] Run `contrib/bump-version v1.6.1_coreos.0`
- [x] Fix cni + flannel #495
- [x] Fix cni + calico #494

### CODE REVIEW

**Comment by mumoshu** [comment]
Self-hosted Calico is not working. See #494

**Comment by mumoshu** [comment]
Found another one: #495

**Comment by camilb** [comment]
Hi @mumoshu I think these changes should also be made to create the minimal roles when RBAC is enabled.
 
   diff --git a/core/controlplane/config/templates/cloud-config-controller b/core/controlplane/config/templates/cloud-config-controller
    index 1c57c4d..f2c3a20 100644
    --- a/core/controlplane/config/templates/cloud-config-controller
    +++ b/core/controlplane/config/templates/cloud-config-controller
    @@ -491,7 +491,7 @@ write_files:
           #!/bin/bash -e
     
           kubectl() {
    -          /usr/bin/docker run --rm --net=host -v /srv/kubernetes/manifests:/srv/kubernetes/manifests {{.HyperkubeImage.RepoWithTag}} /hyperkube kubectl "$@"
    +          /usr/bin/docker run --rm --net=host -v /srv/kubernetes:/srv/kubernetes {{.HyperkubeImage.RepoWithTag}} /hyperkube kubectl "$@"
           }
     
           mfdir=/srv/kubernetes/manifests
    @@ -845,7 +845,7 @@ write_files:
       - path: /srv/kubernetes/rbac/cluster-roles/cluster-admin.yaml
         content: |
             kind: ClusterRole
    -        apiVersion: rbac.authorization.k8s.io/v1alpha1
    +        apiVersion: rbac.authorization.k8s.io/v1beta1
             metadata:
                 name: cluster-admin
             rules:
    @@ -860,7 +860,7 @@ write_files:
       - path: /srv/kubernetes/rbac/cluster-roles/bootstrapped-node.yaml
         content: |
             kind: ClusterRole
    -        apiVersion: rbac.authorization.k8s.io/v1alpha1
    +        apiVersion: rbac.authorization.k8s.io/v1beta1
             metadata:
                 name: bootstrapped-node
             rules:
    @@ -892,7 +892,7 @@ write_files:
       - path: /srv/kubernetes/rbac/cluster-role-bindings/kube-admin.yaml
         content: |
             kind: ClusterRoleBinding
    -        apiVersion: rbac.authorization.k8s.io/v1alpha1
    +        apiVersion: rbac.authorization.k8s.io/v1beta1
             metadata:
               name: kube-admin
             subjects:
    @@ -906,7 +906,7 @@ write_files:
       - path: /srv/kubernetes/rbac/cluster-role-bindings/kube-worker.yaml
         content: |
             kind: ClusterRoleBinding
    -        apiVersion: rbac.authorization.k8s.io/v1alpha1
    +        apiVersion: rbac.authorization.k8s.io/v1beta1
             metadata:
               name: kube-worker
             subjects:
    @@ -921,7 +921,7 @@ write_files:
       - path: /srv/kubernetes/rbac/cluster-role-bindings/system-worker.yaml
         content: |
             kind: ClusterRoleBinding
    -        apiVersion: rbac.authorization.k8s.io/v1alpha1
    +        apiVersion: rbac.authorization.k8s.io/v1beta1
             metadata:
               name: system-worker
             subjects:
    @@ -938,7 +938,7 @@ write_files:
       - path: /srv/kubernetes/rbac/cluster-role-bindings/bootstrapped-node.yaml
         content: |
             kind: ClusterRoleBinding
    -        apiVersion: rbac.authorization.k8s.io/v1alpha1
    +        apiVersion: rbac.authorization.k8s.io/v1beta1
             metadata:
               name: bootstrapped-node
             subjects:
    @@ -954,7 +954,7 @@ write_files:
       - path: /srv/kubernetes/rbac/cluster-roles/kubelet-bootstrap.yaml
         content: |
             kind: ClusterRole
    -        apiVersion: rbac.authorization.k8s.io/v1alpha1
    +        apiVersion: rbac.authorization.k8s.io/v1beta1
             metadata:
               name: kubelet-bootstrap
             rules:
    @@ -971,7 +971,7 @@ write_files:
       - path: /srv/kubernetes/rbac/cluster-role-bindings/kubelet-bootstrap.yaml
         content: |
             kind: ClusterRoleBinding
    -        apiVersion: rbac.authorization.k8s.io/v1alpha1
    +        apiVersion: rbac.authorization.k8s.io/v1beta1
             metadata:
               name: kubelet-bootstrap
             subjects:
    @@ -1076,7 +1076,7 @@ write_files:
               - --tls-private-key-file=/etc/kubernetes/ssl/apiserver-key.pem
               - --client-ca-file=/etc/kubernetes/ssl/ca.pem
               - --service-account-key-file=/etc/kubernetes/ssl/apiserver-key.pem
    -          - --runtime-config=extensions/v1beta1/networkpolicies=true,batch/v2alpha1{{if .Experimental.Plugins.Rbac.Enabled}},rbac.authorization.k8s.io/v1alpha1=true{{ end }}{{if .Experimental.Admission.PodSecurityPolicy.Enabled}},extensions/v1beta1/podsecuritypolicy=true{{ end }}
    +          - --runtime-config=extensions/v1beta1/networkpolicies=true,batch/v2alpha1{{if .Experimental.Plugins.Rbac.Enabled}},rbac.authorization.k8s.io/v1beta1=true{{ end }}{{if .Experimental.Admission.PodSecurityPolicy.Enabled}},extensions/v1beta1/podsecuritypolicy=true{{ end }}
               - --cloud-provider=aws
               livenessProbe:
                 httpGet:
    @@ -1184,7 +1184,7 @@ write_files:
             hostNetwork: true
             volumes:
             {{if .Experimental.DisableSecurityGroupIngress}}
    -        - hostPath: 
    +        - hostPath:
                 path: /etc/kubernetes/additional-configs
               name: additional-configs
             {{end}}

File: model/etcd.go [hunk 1]
```diff
@@ -129,15 +129,15 @@ func (e Etcd) SystemdUnitName() string {
 	return "etcd2.service"
 }
 
-// Version returns the version of etcd (e.g. `2`, `3`, `3.1.3`) to be used for this etcd cluster
+// Version returns the version of etcd (e.g. `2`, `3`, `3.1.4`) to be used for this etcd cluster
 func (e Etcd) Version() EtcdVersion {
+	if e.Cluster.Version == "3" {
+		return "3.1.5"
+	}
 	if e.Cluster.Version != "" {
```
Comment (redbaron) [inline model/etcd.go]: why first check for "3"?
```
  if e.Cluster.Version != "" {
    return e.Cluster.Version
  }
  return "3.1.5"
```

Comment (mumoshu) [inline model/etcd.go]: The intention here is to translate:
```yaml
etcd:
  version: 3
```
to
```yaml
etcd:
  version: 3.1.5
```
for users who doesn't care much about minor/patch versions.
However as kube-aws starts defaulting to etcd3 from now on, we can just default to "3.1.5" when it is omitted, as you've suggested 👍

**Comment by mumoshu** [comment]
E2E is passing against a cluster with/without Calico.

**Pull request merged**